enumerables.rb

```diff
@@ -19,47 +21,84 @@ def my_select
     res
   end
 
-  def my_all?
-    return enum_for(__callee__) unless block_given?
+  # rubocop:disable Metrics/PerceivedComplexity
+  # rubocop:disable Metrics/CyclomaticComplexity
+  def my_all?(cond = nil)
+    if !block_given? && cond.nil?
+      my_each { |i| return false unless i }
+
+    elsif block_given?
+      my_each { |i| return false unless yield(i) }
+
+    elsif cond.is_a? Class
 
-    my_each { |i| return false unless yield(i) }
+      my_each { |i| return false unless i.is_a? cond }
+    elsif cond.is_a? Regexp
+
+      my_each { |i| return false unless i.match cond }
+    else
+      my_each { |i| return false unless i == cond }
+    end
     true
   end
 
-  def my_any?
-    return enum_for(__callee__) unless block_given?
+  def my_any?(cond = nil)
+    if !block_given? && cond.nil?
+      my_each { |i| return true if i }
+
+    elsif block_given?
+      my_each { |i| return true if yield(i) }
+
+    elsif cond.is_a? Class
 
-    my_each { |i| return true if yield(i) }
+      my_each { |i| return true if i.is_a? cond }
+    elsif cond.is_a? Regexp
+
+      my_each { |i| return true if i.match cond }
+    else
+      my_each { |i| return true if i == cond }
+    end
     false
   end
 
-  def my_none?
-    return enum_for(__callee__) unless block_given?
+  # rubocop:enable Metrics/PerceivedComplexity
+  # rubocop:enable Metrics/CyclomaticComplexity
 
-    my_each { |i| return false if yield(i) }
-    true
+  def my_none?(cond = nil)
+    if !block_given? && cond.nil?
+      !my_any?
+
+    elsif block_given?
+
+      !my_any? { yield }
+
+    else
+      !my_any?(cond)
+    end
   end
```

Comment:
`my_none?` identical and returns the same thing as ruby's [`none?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-none-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_none.png) from [Odin](https://github.com/TheOdinProject/curriculum/blob/master/ruby_programming/archive/basic_ruby/project_advanced_building_blocks.md#project-2-enumerable-methods))
- returns true if the block never returns true for all elements
```ruby
print [1, 2, 3].my_none? {|i| i > 4} # should print true but your one throws an error.
```
- returns false if the block ever returns true for all elements.
```ruby
print [1, 2, 3].my_none? {|i| i < 4} # should print false but your one throws an error.
```
- returns false if the block ever returns true for all elements::range
```ruby
print (1...3).my_none? {|i| i < 4} # should print false but your one throws an error.
```

Reply:
Updated method